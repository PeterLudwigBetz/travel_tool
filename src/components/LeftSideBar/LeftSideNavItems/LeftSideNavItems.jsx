import React, { PureComponent, Fragment } from 'react';
import { PropTypes } from 'prop-types';
import LeftSidebarNavItem from './LeftSideNavItem/LeftSideNavItem';
import DropdownItem from './DropdownItems/DropdownItem/DropdownItem';
import './LeftSideNavItems.scss';


class LeftSideNavItems extends PureComponent {

  static propTypes = {
    setActiveNavItem: PropTypes.func.isRequired,
    activeNavItem: PropTypes.object.isRequired,
    navIconsSource: PropTypes.object.isRequired
  };

  static childContextTypes = {
    activeNavItem: PropTypes.object.isRequired,
    setActiveNavItem: PropTypes.func.isRequired
  };

  getChildContext() {
    const {setActiveNavItem, activeNavItem} = this.props;
    return {
      activeNavItem,
      setActiveNavItem
    };
  }

  renderRequestsDropdownItems = () => {
    return (
      <Fragment>
        <DropdownItem link_to="/requests">
          My Requests
        </DropdownItem>
        <DropdownItem link_to="/requests/my-approvals">
          My Approvals
        </DropdownItem>
      </Fragment>
    );
  }

  render() {
    const { navIconsSource } = this.props;

    return (
      <ul>
        <LeftSidebarNavItem isDropdown linkIcons={navIconsSource.requestsIcon} link_to="/requests" text="Requests">
          { this.renderRequestsDropdownItems() }
        </LeftSidebarNavItem>
        <LeftSidebarNavItem linkIcons={navIconsSource.settingsIcon} link_to="/settings" text="Settings" />
      </ul>
    );
  }
}

export default LeftSideNavItems;
