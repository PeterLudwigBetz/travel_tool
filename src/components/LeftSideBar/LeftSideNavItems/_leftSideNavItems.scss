.left-sidebar {
  ul {
    margin: 0;
    padding: 0;

    li {
      box-sizing: border-box;
      display: flex;
      width: 100%;
      height: 48px;
      padding-left: 0;
      outline: none;
      cursor: pointer;

      & > div, & > a {
        text-decoration: none;
        display: flex;
        outline: none;
        height: 100%;
        width: 100%;
      }

      &:not([class~="active"]):hover .left-side-nav-item__nav-text {
        color: black;
      }
    }
  }

  &__dropdown-links {
      height: 21px;
      font-size: 16px;
      font-weight: 500;
      color: #4F4F4F;
      margin-bottom: 25px;
  }

  .active{
    color: #3359DB;
  }

     

}
