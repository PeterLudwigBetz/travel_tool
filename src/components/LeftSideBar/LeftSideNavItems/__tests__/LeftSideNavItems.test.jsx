import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import LeftSideNavItems from '../../LeftSideNavItems/LeftSideNavItems';

describe('<LeftSideNavItems />', () => {
  let wrapper;

  beforeEach(() => {
    const navIconsSource = {
      requestsIcon: {
        active: 'test.svg',
        inactive: 'test.svg'
      },
      settingsIcon: {
        active: 'test.svg',
        inactive: 'test.svg'
      },
    };

    wrapper = mount(
      <MemoryRouter>
        <LeftSideNavItems
          activeNavItem={{}}
          navIconsSource={navIconsSource}
          setActiveNavItem={jest.fn}
        />
      </MemoryRouter>
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('it renders two dropdown items', () => {
    const dropDownItems = wrapper.find('DropdownItem');
    expect(dropDownItems).toHaveLength(2);
  });
});
