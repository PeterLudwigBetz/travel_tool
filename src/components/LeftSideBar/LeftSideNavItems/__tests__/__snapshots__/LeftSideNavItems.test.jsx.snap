// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `undefined`;

exports[`<LeftSideNavItems /> renders correctly 1`] = `
<ul>
  <undefined />
  <undefined />
  <li
    className="left-side-nav-item  inactive"
  >
    <a
      aria-current={null}
      className="nav-link"
      href="/checklist"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="left-side-nav-item__left-icon"
      >
        <img
          alt="icon"
          src="accomodation-grey.svg"
        />
      </div>
      <div
        className="left-side-nav-item__nav-text"
      >
        Trip Planner
      </div>
    </a>
  </li>
  <undefined />
  <li
    className="left-side-nav-item logout-button inactive"
  >
    <a
      aria-current={null}
      className="nav-link"
      href="/logout"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="left-side-nav-item__left-icon"
      >
        <img
          alt="icon"
          src="logout_inactive.svg"
        />
      </div>
      <div
        className="left-side-nav-item__nav-text"
      >
        Logout
      </div>
    </a>
  </li>
</ul>
`;
